src/expr/proof_step.h

```diff
@@ -0,0 +1,65 @@
+/*********************                                                        */
+/*! \file proof_step.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2019 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Proof step enumeration
+ **/
+
+#include "cvc4_private.h"
+
+#ifndef CVC4__EXPR__PROOF_STEP_H
+#define CVC4__EXPR__PROOF_STEP_H
+
+#include "util/safe_print.h"
+
+namespace CVC4 {
+
+/**
+ * An enumeration for proof steps. This enumeration is analogous to Kind for
+ * Node objects. In the documentation below, P:F denotes a ProofNode that
+ * proves formula F.
+ */
+enum class ProofStep : uint32_t
+{
+  //================================================= CORE
```

Comment:
What does CORE stand for?

Reply:
I'll change to "core" (it was stylize like a theory id).

Reply:
But it'd be good to also clarify what does it mean for a rule to be a "core rule"

Reply:
Added